sys/mips/mips/stack_machdep.c

```diff
@@ -156,8 +165,8 @@ stack_save(struct stack *st)
 	if (curthread == NULL)
 		panic("stack_save: curthread == NULL");
 
-	/* XXXRW: Should be pcb_context? */
-	pc = TRAPF_PC(&curthread->td_pcb->pcb_regs);
-	sp = curthread->td_pcb->pcb_regs.sp; // FIXME: use $c11 for CHERI purecap
-	stack_capture(st, pc, sp);
+	pc = (uintptr_t)stack_save;
+	__asm __volatile("move %0, $sp" : "=&r" (sp));
```

Comment:
will __builtin_frame_address() also work, or does this need to be the current SP? 
__builtin_frame_address() will do the right thing for purecap too.

Reply:
PowerPC uses just `__builtin_frame_address(0)`, whilst Arm, Arm64 and RISC-V use both. X86 only uses the frame pointer, but obtained through inline assembly. It's a mess upstream.

Reply:
In the purecap kernel there was already a separate stack_machdep.c and I changed it to use `c11`.  It may be that builtin_frame_address would work.  I haven't tried it.